src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts

```diff
@@ -180,19 +181,37 @@ export class NotebookProviderInfoStore extends Disposable {
 			};
 			const notebookEditorOptions = {
 				canHandleDiff: () => !!this._configurationService.getValue(NotebookSetting.textDiffEditorPreview) && !this._accessibilityService.isScreenReaderOptimized(),
-				canSupportResource: (resource: URI) => resource.scheme === Schemas.untitled || resource.scheme === Schemas.vscodeNotebookCell || this._fileService.hasProvider(resource)
+				canSupportResource: (resource: URI) => {
+					if (resource.scheme === Schemas.vscodeNotebookCellOutput) {
+						const params = new URLSearchParams(resource.query);
+						return params.get('openIn') === 'notebook';
+					}
+					return resource.scheme === Schemas.untitled || resource.scheme === Schemas.vscodeNotebookCell || this._fileService.hasProvider(resource);
+				}
 			};
-			const notebookEditorInputFactory: EditorInputFactoryFunction = ({ resource, options }) => {
-				const data = CellUri.parse(resource);
+			const notebookEditorInputFactory: EditorInputFactoryFunction = async ({ resource, options }) => {
+				let data;
+				if (resource.scheme === Schemas.vscodeNotebookCellOutput) {
+					const outputUriData = CellUri.parseCellOutputUri(resource);
+					if (!outputUriData || !outputUriData.notebook || !outputUriData.cellFragment) {
+						throw new Error('Invalid cell output uri');
+					}
+					data = {
+						notebook: outputUriData.notebook,
+						handle: outputUriData.cellFragment
```

Comment:
`handle` is a number, not sure how `cellFragment` will work here.
The return value of `CellUri.parse` is `{notebook: URI; handle: number}`
Hence both cannot overlap,

Reply:
I think this isn't correct, handle !== fragment.
@rebornix /cc